src/data/list/zip.lean

```diff
@@ -185,6 +185,17 @@ by { rw ←zip_map', congr, exact map_id _ }
 lemma map_prod_right_eq_zip {l : list α} (f : α → β) : l.map (λ x, (f x, x)) = (l.map f).zip l :=
 by { rw ←zip_map', congr, exact map_id _ }
 
+lemma zip_with_comm (f : α → α → β) (comm : ∀ (x y : α), f x y = f y x)
+  (l l' : list α) :
+  zip_with f l l' = zip_with f l' l :=
+begin
```

Comment:
Or perhaps more extremely, you could have::
```suggestion
instance (f : α → α → β) [is_symm_op f] : is_symm_op (zip_with f)  :=
begin
  split,
  intros l l',
```

I don't think `is_symm_op` is well-used though, so perhaps it's best to have both.